### PR TITLE
Fixes #459 - When backend server has changes it causes error on the client side

### LINKED ISSUE #459: When backend server has changes it causes error on the client side
When a breakpoint is set, and the user keeps a debug session listening in the background the client periodically checks the Cloud Debugger backend for changes with a hanging get using a wait token.
If the breakpoint is hit, the backend will return an HTTP 409 to the client to notify it about results being ready to query. When this 409 is received due to recent changes in how we handle errors in requests, it was also handled as error this showing an IDE error notification to the user, instead of simply suppressing it.

### PR DESCRIPTION
**Pull request by akerekes**
@etanshaul @joaoandremartins @elharo PTAL

### CODE REVIEW

File: google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java [hunk 1]
```diff
@@ -169,6 +169,12 @@ void pollForChanges(@NotNull final CloudDebugProcessState state) {
     }
     catch(SocketTimeoutException sto) {
       //noop, this is expected behavior.
+    } catch (GoogleJsonResponseException e) {
+      // HTTP 409 is expected when backend responds to "hanging get" either for timeout or because
+      // a result is available (which will be retrieved via the subsequent query)
+      if (e.getStatusCode() != 409) {
+        handleBreakpointQueryError(state, e);
```
Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java]: Should we log something on a debug/info level just to inform a 409 was received?

Comment (elharo) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java]: Let's not. Logs are hard enough to parse without a lot fo messages from code that ios doing exactly what we expect it to do.

Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java]: Also, just so it's clearer to whoever is reading this, maybe add more to the comments explaining the "hanging get" as well as what is meant by "query".

File: google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java [hunk 2]
```diff
@@ -169,6 +169,12 @@ void pollForChanges(@NotNull final CloudDebugProcessState state) {
     }
     catch(SocketTimeoutException sto) {
       //noop, this is expected behavior.
+    } catch (GoogleJsonResponseException e) {
+      // HTTP 409 is expected when backend responds to "hanging get" either for timeout or because
+      // a result is available (which will be retrieved via the subsequent query)
+      if (e.getStatusCode() != 409) {
```
Comment (elharo) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java]: named constant HttpURLConnection.HTTP_CONFLICT

File: google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java (hunk 1 above)
Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java]: @elharo that's why I suggested to log it on debug -- it doesn't clutter >=info logs.

We expect 409's to be returned -- however, there might be an issue with the server not returning any 409. Having 409's logged on the client would help debug that scenario.

Comment (elharo) [inline google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java]: YAGNI

**Comment by akerekes** [comment]
@etanshaul @joaoandremartins @elharo PTAL

**Comment by etanshaul** [comment]
lgtm

**Comment by joaoandremartins** [comment]
Aside from the logging issue, LGTM.

**Comment by elharo** [comment]
lgtm

**Pull request merged**